yt/fields/derived_field.py

```diff
@@ -34,6 +35,23 @@ def NullFunc(field, data):
     raise YTFieldNotFound(field.name)
 
 
+def DeprecatedFunc(ret_field, func, since, removal):
+    def _DeprecatedFunc(field, data):
+        if data.ds.fields_detected:
```

Comment:
We don't want to emit deprecation messages about fields while the initial field detection stage is going on (which calls every field to see if they work)--because then one could get many, many messages for fields that one doesn't care about. So this makes sure we've already done field detection.